app/src/main/java/com/example/househomey/SignUpFragment.java

```diff
@@ -0,0 +1,247 @@
+    private FirebaseAuth auth;
+    private String username;
+    private String email;
+    private String password;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_user_signup, container, false);
+
+        // Initialize Firebase Auth
+        auth = FirebaseAuth.getInstance();
+
+        usernameEdittext = rootView.findViewById(R.id.signup_username);
+        emailEdittext = rootView.findViewById(R.id.signup_email);
+        passwordEdittext = rootView.findViewById(R.id.signup_password);
+        confirmPasswordEdittext = rootView.findViewById(R.id.signup_confirm_password);
+        signupButton = rootView.findViewById(R.id.signup_button);
+
+        TextView signinRedirectMessage = rootView.findViewById(R.id.signin_redirect_message);
+        TextView signinRedirect = rootView.findViewById(R.id.signin_redirect);
+
+        signupButton.setText("Register");
+        signinRedirectMessage.setText("Already have an account?");
+        signinRedirect.setText(R.string.sign_in_string);
+
+        signupButton.setOnClickListener(v -> {
+            signupButton.setBackgroundResource(R.drawable.signin_button_clicked);
+            signupButton.setTextColor(getResources().getColor(R.color.creme, rootView.getContext().getTheme()));
+            new Handler(Looper.getMainLooper()).postDelayed(() -> {
+                signupButton.setBackgroundResource(R.drawable.signin_button_unclicked);
+                signupButton.setTextColor(getResources().getColor(R.color.brown, rootView.getContext().getTheme()));
+            }, 150);
+
+            collRef = FirebaseFirestore.getInstance().collection("user");
+            if (confirmPassword()) {
+                collRef.document(username).get().addOnSuccessListener(task -> {
+                    if (task.getData() == null) {
+                        UserProfileChangeRequest profileChangeRequest =
+                                new UserProfileChangeRequest.Builder().setDisplayName(username).build();
+                        auth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(task1 -> {
+                            if (task1.isSuccessful()) {
+                                FirebaseUser user = auth.getCurrentUser();
+                                if (user == null) {
+                                    emailEdittext.setError("database error");
+                                    usernameEdittext.setError("database error");
+                                } else {
+                                    user.updateProfile(profileChangeRequest).addOnCompleteListener(task12 -> {
+                                        if (task12.isSuccessful()) {
+                                            Map<String, Object> fields = new HashMap<>();
+                                            fields.put("email", user.getEmail());
```

Comment:
Do we need this? I think we should be able to call FirebaseAuth.getInstance().getCurrentUser().getEmail() at any time to get that.

Reply:
Its needed for the sign in. Since users are signed out before we call it. There is not always a user currently signed in.

Reply:
Can you point out which line needs access to this? I don't see why we would need it.

Reply:
if (document != null) {
                        String email = (String) document.get("email");
                        if (email == null) {

Right here we grab the email from the document since there is no signed in user when the app first starts